File: core/src/com/mygdx/game/characters/Captain.java
package com.mygdx.game.characters;

import com.badlogic.gdx.math.Vector3;
import com.mygdx.game.characters.State.Stand;
import com.mygdx.game.characters.State.State;
import com.mygdx.game.core.RenderObject;

/**
 * Created by Alan on 3/29/2016.
 */
public class Captain extends GameCharacter {

    @Override
    public void load() {
        RenderObject ro = new CaptainRenderObject(this);
        center = new Vector3(20,20,0);
        state = new Stand(this);
        walkSpeed = .3f;
//        ai = new RandomWalkAI(this);
        super.load();
    }

    @Override
    public void render() {

//        ai.update();


        super.render();
    }
}

File: web/src/characters/Captain.ts
import Phaser from 'phaser';
import { GameCharacter } from './GameCharacter';
import { RandomTravelAI } from './ai/RandomTravelAI';

export class Captain extends GameCharacter {
  constructor(scene: Phaser.Scene) {
    super(scene, 'captain', { stand: 'captain_stand', run: 'captain_run' });
    this.walkSpeed = 0.3;
    this.center = { x: 20, y: 20 };
    this.ai = new RandomTravelAI(this);
  }
}
